cmd/abapAddonAssemblyKitCheckPV.go

```diff
@@ -0,0 +1,101 @@
+package cmd
+
+import (
+	"encoding/json"
+
+	abapbuild "github.com/SAP/jenkins-library/pkg/abap/build"
+	"github.com/SAP/jenkins-library/pkg/abaputils"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+func abapAddonAssemblyKitCheckPV(config abapAddonAssemblyKitCheckPVOptions, telemetryData *telemetry.CustomData, cpe *abapAddonAssemblyKitCheckPVCommonPipelineEnvironment) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	client := piperhttp.Client{}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runAbapAddonAssemblyKitCheckPV(&config, telemetryData, &client, cpe, abaputils.ReadAddonDescriptor)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapAddonAssemblyKitCheckPV(config *abapAddonAssemblyKitCheckPVOptions, telemetryData *telemetry.CustomData, client piperhttp.Sender,
+	cpe *abapAddonAssemblyKitCheckPVCommonPipelineEnvironment, readAdoDescriptor abaputils.ReadAddonDescriptorType) error {
+	var addonDescriptorFromCPE abaputils.AddonDescriptor
+	json.Unmarshal([]byte(config.AddonDescriptor), &addonDescriptorFromCPE)
+	addonDescriptor, err := readAdoDescriptor(config.AddonDescriptorFileName)
+	addonDescriptor = combineYAMLProductWithCPERepositories(addonDescriptor, addonDescriptorFromCPE)
+	if err != nil {
+		return err
+	}
+	conn := new(abapbuild.Connector)
+	conn.InitAAKaaS(config.AbapAddonAssemblyKitEndpoint, config.Username, config.Password, client)
+
+	var p pv
+	p.init(addonDescriptor, *conn)
+	err = p.validate()
+	if err != nil {
+		return err
+	}
+	p.copyFieldsToRepo(&addonDescriptor)
+	log.Entry().Info("Write the resolved version to the CommonPipelineEnvironment")
+	toCPE, _ := json.Marshal(addonDescriptor)
+	cpe.abap.addonDescriptor = string(toCPE)
+	return nil
+}
+
+func combineYAMLProductWithCPERepositories(addonDescriptor abaputils.AddonDescriptor, addonDescriptorFromCPE abaputils.AddonDescriptor) abaputils.AddonDescriptor {
```

Comment:
You should be able to reuse this helper function from the previous PR as it is in the same package "cmd".

(I'm not sure, but maybe you have to start the function name with a capital letter)